HelloDI/src/HelloDI.Tests/SalutationTests.cs

```diff
@@ -25,5 +26,16 @@ public void ExclaimWillWriteCorrectMessageToMessageWriter()
                 expected: "Hello DI!",
                 actual: writer.WrittenMessage);
         }
+
+        [Fact]
+        public void NullWriterInConstructorThrowsException()
+        {
+            // Arrange
+            Action action = () => new Salutation(writer: null);
+
+            // Assert
+            var exception = Assert.Throws<ArgumentNullException>(action);
+            Assert.Equal("Value cannot be null.\r\nParameter name: writer", exception.Message);
```

Comment:
I would suggest removing the last `Assert.Equal`, because this check makes the test fragile; the test will fail when ran on anything but an English version of Windows.

I'm also interested in what @ploeh has to say about this.

Reply:
As a general rule, I don't assert that strings have specific values. As @dotnetjunkie writes, it'll just make the test fragile. This'll mean that every time you change the string in your production code, you also have to change the string in the test. Where's the value in that?

(To be clear, I wrote _as a general rule_. If you're testing a function where the main purpose is to produce and format a string in a particular way, then it may not apply.)